R/redist.R

```diff
@@ -707,389 +798,267 @@ redist.combine <- function(savename, nsims, nloop, nthin, nunits, temper = 0
-        segout <- sumstat
-    }
+    ## Run algorithm
+    algout <- swMH(aList = preprocout$data$adjlist,
+                   cdvec = cds,
+                   cdorigvec = preprocout$data$initcds,
+                   popvec = preprocout$data$popvec,
+                   grouppopvec = preprocout$data$grouppopvec,
+                   areas_vec = preprocout$data$areasvec,
+                   county_membership = preprocout$data$countymembership,
+                   borderlength_mat = preprocout$data$borderlength_mat,
+                   nsims = nsims,
+                   eprob = eprob,
+                   pct_dist_parity = preprocout$params$pctdistparity,
+                   beta_sequence = preprocout$params$betaseq,
+                   beta_weights = preprocout$params$betaweights,
+                   ssdmat = preprocout$data$ssdmat,
+                   lambda = lambda,
+                   beta = preprocout$params$beta,
+                   weight_population = weightpop,
+                   weight_compact = weightcompact,
+                   weight_segregation = weightseg,
+                   weight_similar = weightsimilar,
+                   weight_countysplit = weightcountysplit,
+                   adapt_beta = preprocout$params$temperbeta,
+                   adjswap = preprocout$params$adjswaps,
+                   exact_mh = exact_mh,
+                   adapt_lambda = adapt_lambda,
+                   adapt_eprob = adapt_eprob,
+                   compactness_measure = compactness_metric)
     
-    ## Logit transform
-    if(logit){
-        if(class(segout) == "mcmc"){
-            segout <- log(segout / (1 - segout))
-        }else if(class(segout) == "mcmc.list"){
-            for(i in 1:length(segout)){
-                segout[[i]] <- log(segout[[i]] / (1 - segout[[i]]))
-            }
-        }
-    }
-
-    ##################
-    ## Create plots ##
-    ##################
-    if(plot == "trace"){
-        if(!is.null(savename)){
-            pdf(file = paste(savename, ".pdf", sep = ""))
-        }
-        traceplot(segout)
-        if(!is.null(savename)){
-            dev.off()
-        }
-    }
-    if(plot == "autocorr"){
-        if(!is.null(savename)){
-            pdf(file = paste(savename, ".pdf", sep = ""))
-        }
-        autocorr.plot(segout, lag.max = 50)
-        if(!is.null(savename)){
-            dev.off()
-        }
-    }
-    if(plot == "densplot"){
-        if(!is.null(savename)){
-            pdf(file = paste(savename, ".pdf", sep = ""))
-        }
-        densplot(segout)
-        if(!is.null(savename)){
-            dev.off()
-        }
-    }
-    if(plot == "mean"){
-        if(!is.null(savename)){
-            pdf(file = paste(savename, ".pdf", sep = ""))
-        }
-        cumuplot(segout, probs = .5, type = "l", lty = 1)
-        if(!is.null(savename)){
-            dev.off()
-        }
+} # ends if algorithm==mcmc
+  if(algorithm == 'anneal'){

```

Comment:
You shouldn't need a conditional statement here, since it can all happen inside of swMH.cpp. If you take a look at the dev branch, it's conditional on a variable called `std::string adapt_beta`. You could use this instead and call it `algorithm` to have it match the argument on the `redist.mcmc()` wrapper level.

Reply:
And sorry, I am still learning the Github collaborator tools. There are a few more comments I left in the pull request, let me know if you can't find them.